src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java

```diff
@@ -1005,4 +1007,67 @@ public Connection getConnection() {
   public SnowflakeInternalStage getInternalStage() {
     return this.internalStage;
   }
+
+  @Override
+  public boolean updateStreamingChannelOffsetToken(
```

Comment:
Actually, can we just use this function to check whether the new channel exists instead of making the REST API call? If yes, then everything in PART-2 can be discard

Reply:
Thats a good point.. we can use sys function to check if channel exist or not too. 

so then it looks like we will use two sys calls ..

Reply:
actually I just thought about this. We have to have part-2. you cant use this sys func to check if channel exists or not. in doing so you will lose a legit offset token in case a channel exists.

Reply:
Isn't the new offset token be passed in by Kafka valid? so we can throw away the legit offset token of the existing channel anyways